snooty/postprocess.py

```diff
@@ -506,6 +509,9 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             self.current_depth += 1
             return
 
+        if isinstance(node, n.Directive):
+            self.scanned_pattern.append((node.name, node.options))
```

Comment:
Instead of appending for every directive, would it make sense to append only when the desired directive name is found? That could help reduce any unwanted directives